lib/src/component/fastly_abi.rs

```diff
@@ -0,0 +1,16 @@
+use {
+    super::fastly::api::fastly_abi,
+    super::FastlyError,
+    crate::{error::Error, session::Session, wiggle_abi::ABI_VERSION},
+};
+
+#[async_trait::async_trait]
+impl fastly_abi::Host for Session {
+    async fn init(&mut self, abi_version: u64) -> Result<(), FastlyError> {
+        if abi_version != ABI_VERSION {
```

Comment:
I know there's been discussion of removing this entirely, and switching to a wit-based interface would make it somewhat unnecessary. We could remove it here, and implement it entirely in the adapter?

Reply:
That would be lovely, if it wouldn't be too hard. Since the adapter (and the thing it's adapting) is the thing that cares about the version, rather than Viceroy proper, in this case.

Reply:
I've removed this from compute.wit, and implemented it over in the adapter instead.